apps/site/components/Common/Button/index.tsx

```diff
@@ -17,17 +22,46 @@ const Button: FC<ButtonProps> = ({
   href = undefined,
   children,
   className,
+  onClick,
   ...props
-}) => (
-  <Link
-    role="button"
-    href={disabled ? undefined : href}
-    aria-disabled={disabled}
-    className={classNames(styles.button, styles[kind], className)}
-    {...props}
-  >
-    {children}
-  </Link>
-);
+}) => {
+  //to handle the keyboard interactions, specifically for Enter and Space keys
+  const handleKeyDown = (e: KeyboardEvent<HTMLAnchorElement>) => {
+    if (disabled) return;
+
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      if (onClick) {
+        onClick(e as unknown as MouseEvent<HTMLAnchorElement>);
+      }
+    }
+  };
+
+  // to manage mouse click events for the component, providing behavior consistent with the disabled state
+  const handleClick = (e: MouseEvent<HTMLAnchorElement>) => {
```

Comment:
I feel that calling it onClickHandler sounds more concise with the naming of the event itself